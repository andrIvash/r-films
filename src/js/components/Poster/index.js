// @flow
import React from 'react';
import type { PosterData } from '../../flow-types.js';

type Props = {
 data: PosterData
}

const Poster = (props: Props) => (
  <div className='poster'>
    <div className='poster__img-wrap'>
      <img alt='file poster' className='poster__img' src={props.data.poster_path} />
    </div>
    <div className='poster__info film'>
      <div className='film__header'>
        <h1 className='film__title'>{props.data.title}</h1>
        <div className='film__score'>{props.data.vote_average}</div>
      </div>
      <h5 className='film__subtitle'>{props.data.tagline}</h5>
      <div className='film__info'>
        <div className='film__year'>{props.data.release_date.substr(0, 4)}</div>
        <div className='film__latency'>{props.data.runtime ? `${props.data.runtime} min` : ''}</div>
      </div>
      <p className='film__desq'>{props.data.overview}
      </p>
    </div>
  </div>
);

export default Poster;
