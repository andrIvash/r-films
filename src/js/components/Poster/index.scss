.poster {
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  color: $color-layout-dark;
  margin-bottom: rem(20px);
}

.poster__img-wrap {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 30%;
  height: 100%;
  @include tablet() {
    width: 100%;
  }
  .poster__img {
    width: 100%;
    max-width: rem(200px);
    height: auto;
  }
}

.poster__info {
  width: 65%;
  @include tablet() {
    width: 100%;
  }
}

.film__header {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: rem(10px);
  font-family: $sign-font;

  @include tablet() {
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: center;
  }

  .film__title {
    color: $color-logo;
    width: 80%;
  }

  .film__score {
    display: flex;
    justify-content: center;
    align-items: center;
    width: rem(40px);
    height: rem(40px);
    margin-left: rem(20px);
    border-radius: 50%;
    border: 2px solid $color-layout-dark;
    font-size: rem(14px);  
  }
}

.film__subtitle {
  @include tablet() {
    text-align: center;
  }
}

.film__info {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: rem(20px);

  @include tablet() {
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: center;
  }
}
.film__year,
.film__latency {
  display: inline-block;
  font-family: $sign-font;
  font-size: rem(13px);
  color: $color-white;
  
  @include tablet() {
    display: inline-block;
    width: 100%;
    margin-left: 0;
  }
}

.film__latency {
  margin-left: rem(20px);

  @include tablet() {
    margin-left: 0;
  }
}

.film__desq {
  margin-bottom: 0;
  font-family: $base-font;
  font-size: rem(16px);
}
