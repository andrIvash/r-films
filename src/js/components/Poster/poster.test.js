import React from 'react';
import Poster from './index';

const props = {
  data: {
    vote_average: 'http://test',
    release_date: '1924',
    title: 'title',
    tagline: 'tagline',
    runtime: 100,
  },
};

describe('Poster', () => {
  it('should be defined', () => {
    expect(Poster).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = shallow(
      <Poster {...props} /> );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render write values ', () => {
    const wrapper = shallow(
      <Poster {...props} /> );
    expect(wrapper.find('.film__title').text()).toBe('title');
    expect(wrapper.find('.film__year').text()).toBe('1924');
  });
});
