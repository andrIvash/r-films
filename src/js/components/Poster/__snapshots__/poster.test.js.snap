// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster should render correctly 1`] = `
<div
  className="poster"
>
  <div
    className="poster__img-wrap"
  >
    <img
      alt="file poster"
      className="poster__img"
    />
  </div>
  <div
    className="poster__info film"
  >
    <div
      className="film__header"
    >
      <h1
        className="film__title"
      >
        title
      </h1>
      <div
        className="film__score"
      >
        http://test
      </div>
    </div>
    <h5
      className="film__subtitle"
    >
      tagline
    </h5>
    <div
      className="film__info"
    >
      <div
        className="film__year"
      >
        1924
      </div>
      <div
        className="film__latency"
      >
        100 min
      </div>
    </div>
    <p
      className="film__desq"
    />
  </div>
</div>
`;
